Added Angular guide

engineering-this opened the pull request:
Added angular guide, which partially solves https://github.com/ckeditor/ckeditor-dev/issues/2920.

mlewand commented on docs/guide/dev/integration/angular/README.md:
```diff
@@ -0,0 +1,241 @@
+<info-box info=""> This feature is provided through the <a href="https://www.npmjs.com/package/ckeditor4-angular"><code>ckeditor4-angular</code> npm package</a>.
+</info-box>
+
+CKEditor offers native Angular integration through `CKEditor` Angular component. It provides deep integration of CKEditor and Angular that lets you use the native features of CKEditor inside Angular component. The `CKEditor` Angular component is compatible with Angular versions 2.0+.
```
```suggestion
CKEditor offers native Angular integration through `CKEditor` Angular component. It provides deep integration of CKEditor and Angular that lets you use the native features of CKEditor inside Angular component. The `CKEditor` Angular component is compatible with Angular versions 2.0 and later.
```

mlewand commented on docs/guide/dev/integration/angular/README.md:
````diff
@@ -0,0 +1,241 @@
+<ckeditor [editorUrl]="https://your-website.example/ckeditor/ckeditor.js"></ckeditor>
+```
+
+Note that attribute has to be assigned **before first component is initialized**.
````
```suggestion
Note that attribute has to be assigned **before the first component is initialized**.
```

mlewand commented on docs/guide/dev/integration/angular/README.md:
```diff
@@ -0,0 +1,241 @@
+    ...
+} )
+export class MyComponent {
+    public Editor = ClassicEditor;
```
I don't understand this code, ClassicEditor is not defined.

mlewand commented on docs/guide/dev/integration/angular/README.md:
```diff
@@ -0,0 +1,241 @@
+
+## Choosing Editor Type
+
+By default `CKEditor` Angular component creates {@link guide/dev/framed/README classic editor}. To create {@link guide/dev/inline/README inline editor} `type` property with value of `inline` must be added:
```
From what I know, currently there's no support for classic editor in our angular integration. Is this section relevant to this integration?

engineering-this commented:
Pushing changes:
- Rephrased _editor types_ section, to reflect actual supported editor types.
- Corrected few code samples, removed redundant declarations/attributes.
- Corrected other requested issues, grammar, etc.

mlewand commented on docs/guide/dev/integration/angular/README.md:
```diff
@@ -0,0 +1,245 @@
+
+### `change`
+
+Fires when the content of the editor has changed. In contrast to `change` - only emits when data really changed thus can be successfully used with `[data]` and two way `[(data)]` binding.
```
"Fires when the content of the editor has changed." is duplicated **three** times, this makes it extremely redundant.

mlewand replied:
How `change` event can work "In contrast to `change` "? `change` is the event that is described here.

mlewand commented on docs/guide/dev/integration/angular/README.md:
```diff
@@ -0,0 +1,245 @@
+
+### `dataChange`
+
+Fires when the content of the editor has changed. In contrast to `change` - only emits when data really changed thus can be successfully used with `[data]` and two way `[(data)]` binding.
```
This is an exact copy of "Fires when the content of the editor has changed. In contrast to change - only emits when data really changed thus can be successfully used with [data] and two way [(data)] binding." from a previous section.

engineering-this replied:
I left this section as it correct and updated `change`.

mlewand commented on docs/guide/dev/integration/angular/README.md:
```diff
@@ -0,0 +1,245 @@
+
+### `focus`
+
+Fires when the editing view of the editor is focused. It corresponds with the `editor#focus` https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editor.html#event-focus event.
```
Most common name for this is editable, let's stick to it.

```suggestion
Fires when the editor's editable is focused. It corresponds with the `editor#focus` https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editor.html#event-focus event.
```

Same change for the blur event.

mlewand replied:
It should be a link, instead there's a event name followed by a full URL. Same thing for blur.

mlewand commented on docs/guide/dev/integration/angular/README.md:
```diff
@@ -0,0 +1,245 @@
+
+## Types
+
+CKEditor types can be installed from [@types/ckeditor](https://www.npmjs.com/package/@types/ckeditor). Note, they aren't developed by the CKEditor Team, so we can't guarantee that are up to date.
```
```suggestion
CKEditor types can be installed from [@types/ckeditor](https://www.npmjs.com/package/@types/ckeditor). Note, they aren't maintained by the CKEditor Team, so we can't guarantee that are up to date.
```

engineering-this commented on docs/guide/dev/integration/angular/README.md:
````diff
@@ -0,0 +1,245 @@
+
+```html
+<script src="custom-ckeditor.js"></script>
+<script src="main.js"></script>
````
I mean bundled Angular package, as it has to be loaded after the `ckeditor.js`. What do you think, about adding a comment:

```suggestion
<script src="main.js"></script><!–– Your bundled Angular package ––>
```
or
```suggestion
<script src="your-app-bundle.js"></script>
```
or just drop this line?

mlewand replied:
Both propositions look good. I think the second is cleaner 👌

engineering-this commented:
Additionally I've altered some links, to use `{@linkapi}` or `{@link}`.

mlewand commented on docs/guide/dev/integration/angular/README.md:
````diff
@@ -0,0 +1,235 @@
+By default `CKEditor` Angular component loads [Standard Preset](https://ckeditor.com/docs/ckeditor4/latest/examples/standardpreset.html) of the latest CKEditor release from <a href="https://cdn.ckeditor.com/">CDN</a> while creating the first editor. This behavior can be altered by setting the value of `editorUrl` attribute in template to point to desired CKEditor location:
+
+```html
+<ckeditor [editorUrl]="https://your-website.example/ckeditor/ckeditor.js"></ckeditor>
````
Setting `editorUrl` attribute like that will throw exception:

```
compiler.js:15859 Uncaught Error: Quotes are not supported for evaluation!
        Statement: //cdn.ckeditor.com/4.5.5/standard-all/ckeditor.js located at ng:///AppModule/AppComponent.html@3:11
```

mlewand replied:
Should be without brackets.

mlewand commented on docs/guide/dev/integration/angular/README.md:
````diff
@@ -0,0 +1,235 @@
+```html
+<ckeditor
+	data="<p>Editor' content</p>"
+	[config]="{ { toolbar: [ [ 'Bold' ] ] } }"
````
Value of `[config]` throws parsing error to the console.

engineering-this commented on docs/guide/dev/integration/angular/README.md:
```diff
@@ -0,0 +1,235 @@
+
+### `change`
+
+Fires when the content of the editor has changed. It corresponds with the {@linkapi CKEDITOR.editor#change `editor#change`} event. For performance reasons this event may be called even when data didn't really changed.
```
Components `change` fully matches https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editor.html#event-change the rationale comes from our docs.

engineering-this commented:
- Corrected code samples, they are all working correctly now.
- Fixed broken link.
- Removed ellipsis symbol from html samples, they didn't look good there, and `<ckeditor></ckeditor> work without any extra attributes anyway.

mlewand commented on docs/guide/dev/integration/angular/README.md:
````diff
@@ -0,0 +1,234 @@
+
+``` typescript
+
+import { CKEditor4 } from "@ckeditor/ckeditor4-angular/ckeditor";
````
We use single quotes for js strings.

mlewand commented on docs/guide/dev/integration/angular/README.md:
```diff
@@ -0,0 +1,234 @@
+
+### Note
+
+Callbacks bound to each of events, are called with {@linkapi CKEDITOR.eventInfo `eventInfo`}. For importing and usage of `EventInfo` interface see [**`change`** example](#change).
```
Not each event, as [`dataChange` is fired with a different signature](https://github.com/cksource/ckeditor4-angular/blob/3bfa435/src/ckeditor/ckeditor.component.ts#L293) it gets a plain string.

mlewand replied:
I think it should be changed in the upstream, that is `ckeditor4-angular` integration.
